README.rst

```diff
@@ -12,77 +12,83 @@ smart_open — utils for streaming large files in Python
   ...     for line in fin:
-  ...         print(line.decode('utf8'))
-  ...     fin.seek(0)  # seek to the beginning
-  ...     b1000 = fin.read(1000)  # read 1000 bytes
-
-  >>> # stream from HDFS
-  >>> for line in smart_open('hdfs://user/hadoop/my_file.txt', encoding='utf8'):
-  ...     print(line)
+  ...         print(line.decode('utf-8'), end='')
+  ...         break
+  ...     offset = fin.seek(0)  # seek to the beginning
+  ...     print(fin.read(4))
+  User-Agent: *
+  b'User'
 
   >>> # stream from HTTP
-  >>> for line in smart_open('http://example.com/index.html'):
-  ...     print(line)
-
-  >>> # stream from WebHDFS
-  >>> for line in smart_open('webhdfs://host:port/user/hadoop/my_file.txt'):
-  ...     print(line)
-
-  >>> # stream content *into* S3 (write mode):
-  >>> with smart_open('s3://mybucket/mykey.txt', 'wb') as fout:
-  ...     for line in [b'first line\n', b'second line\n', b'third line\n']:
-  ...          fout.write(line)
-
-  >>> # stream content *into* HDFS (write mode):
-  >>> with smart_open('hdfs://host:port/user/hadoop/my_file.txt', 'wb') as fout:
-  ...     for line in [b'first line\n', b'second line\n', b'third line\n']:
-  ...          fout.write(line)
-
-  >>> # stream content *into* WebHDFS (write mode):
-  >>> with smart_open('webhdfs://host:port/user/hadoop/my_file.txt', 'wb') as fout:
-  ...     for line in [b'first line\n', b'second line\n', b'third line\n']:
-  ...          fout.write(line)
-
-  >>> # stream using a completely custom s3 server, like s3proxy:
-  >>> for line in smart_open('s3u://user:secret@host:port@mybucket/mykey.txt', 'rb'):
-  ...    print(line.decode('utf8'))
-
-  >>> # you can also use a boto.s3.key.Key instance directly:
-  >>> key = boto.connect_s3().get_bucket("my_bucket").get_key("my_key")
-  >>> with smart_open(key, 'rb') as fin:
-  ...     for line in fin:
-  ...         print(line.decode('utf8'))
- 
-  >>> # Stream to Digital Ocean Spaces bucket providing credentials from boto profile
-  >>> with smart_open('s3://bucket-for-experiments/file.txt', 'wb', endpoint_url='https://ams3.digitaloceanspaces.com', profile_name='digitalocean') as fout:
-  ...     fout.write(b'here we stand')
+  >>> for line in open('http://example.com/index.html'):
+  ...     print(line, end='')
+  ...     break
+  <!doctype html>
+
+Other examples of URLs that ``smart_open`` accepts::
+
+    s3://my_bucket/my_key
+    s3://my_key:my_secret@my_bucket/my_key
+    s3://my_key:my_secret@my_server:my_port@my_bucket/my_key
+    hdfs:///path/file
+    hdfs://path/file
+    webhdfs://host:port/path/file
+    ./local/path/file
+    ~/local/path/file
+    local/path/file
+    ./local/path/file.gz
+    file:///home/user/file
+    file:///home/user/file.bz2
+    [ssh|scp|sftp]://username@host//path/file
+    [ssh|scp|sftp]://username@host/path/file
+    file:///home/user/file.xz
+
+For detailed API info, see the online help:
+
+.. code-block:: python
+
+    help('smart_open')
```

Comment:
Are you on the branch of this PR? If not, then the command won't show you anything useful, because the previous way of exposing the package's functionality was broken.

This is part of what you would see if you have this branch checked out:

```
$ python -c 'help("smart_open")' | head -n 40
Help on package smart_open:

NAME
    smart_open

DESCRIPTION
    Utilities for streaming to/from several file-like data storages: S3 / HDFS / local
    filesystem / compressed files, and many more, using a simple, Pythonic API.
    
    The streaming makes heavy use of generators and pipes, to avoid loading
    full file contents into memory, allowing work with arbitrarily large files.
    
    The main functions are:
    
    * `open()`, which opens the given file for reading/writing
    * `s3_iter_bucket()`, which goes over all keys in an S3 bucket in parallel
    * `register_compressor()`, which registers callbacks for transparent compressor handling

PACKAGE CONTENTS
    doctools
    hdfs
    http
    s3
    smart_open_lib
    ssh
    tests (package)
    webhdfs

FUNCTIONS
    open(uri, mode='r', buffering=-1, encoding=None, errors=None, newline=None, closefd=True, opener=None, ignore_ext=False, transport_params={})
        Open the URI object, returning a file-like object.
        
            The URI is usually a string in a variety of formats:
        
            1. a URI for the local filesystem: `./lines.txt`, `/home/joe/lines.txt.gz`, `file:///home/joe/lines.txt.bz2`
            2. a URI for HDFS: `hdfs:///some/path/lines.txt`
            3. a URI for Amazon's S3 (can also supply credentials inside the URI):
               `s3://my_bucket/lines.txt`, `s3://my_aws_key_id:key_secret@my_bucket/lines.txt`
        
            The URI may also be one of:
```

Reply:
No, I was on `master`. Thanks!